custom/icds_reports/tests/outputs/ccs_record_monthly_2017-04-01_sorted.csv

```diff
@@ -3,7 +3,7 @@ a1,0f2ecc69-9f03-46e1-95fc-f28df2d5951d,2017-04-01,280,other,1,1,,0,0,0,0,0,0,0,
 a1,1377feed-291a-4eb4-ab61-45ef23f41716,2017-04-01,280,other,1,1,,0,0,0,0,0,0,0,0,,0,0,0,0,0,1,0,0,0,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,,0,0,0,0,0,2016-05-01,0,obc,no,no,yes,,,,,,,,,,,,,,,,2017-08-08,,,,,,,,,,0,,,,,,,2017-04-12,,1994-01-08,0,3ecb07a8-abb5-403e-9fce-44b4f8a045b3
 a1,32890569-edc5-43ea-94df-430945afef1d,2017-04-01,472,other,1,1,,0,0,0,0,0,0,0,0,,0,0,0,0,0,1,0,0,0,,0,0,0,0,0,0,0,0,0,0,0,0,0,1,,0,0,0,0,0,2016-05-01,0,obc,no,no,yes,,,,,,,,,,,,,,,,2017-08-08,,,,,,,,,,0,,,,,,,2017-04-12,,1978-01-16,0,0ee12d13-6321-48e0-86d7-426cc7c58b96
 a1,346ad0e7-210c-4b6f-9659-534f473922d1,2017-04-01,340,other,1,1,,0,0,0,0,0,0,0,0,,0,0,0,0,0,1,0,0,0,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,,0,0,0,0,0,2016-05-01,0,obc,no,no,yes,,,,,,,,,,,,,,,,2017-08-08,,,,,,,,,,0,,,,,,,2017-04-12,,1989-01-10,0,d1bd8a2f-f874-4a08-aeaf-41a65ec7b5f1
-a1,387085d6-7427-48c3-aee3-ff12819def46,2017-04-01,316,lactating,1,1,,20,1,0,0,0,0,0,0,,0,0,0,0,0,1,0,0,0,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,,0,0,1,1,0,2017-03-01,0,obc,no,no,yes,,,,,,,,,,,,,,,,2017-08-08,,,,,,,,,,1,,,,,,,2017-04-12,,1991-01-10,0,a9bc2bb8-0f4f-4e21-a51e-39369d83fb1c
+a1,387085d6-7427-48c3-aee3-ff12819def46,2017-04-01,316,lactating,1,1,,20,1,0,0,0,0,0,0,,0,0,0,0,0,1,0,0,0,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,,0,0,1,1,0,2017-03-01,0,obc,no,no,yes,,,,,,,,,,,,,,,,2017-08-08,,,,,,,,,,1,,,2017-03-04,,,,2017-04-12,2017-03-04,1991-01-10,0,a9bc2bb8-0f4f-4e21-a51e-39369d83fb1c
```

Comment:
bp_date  and home_visit_date contains the same info. Not sure why home_visit_date was added.
Though just adding those date info

Reply:
A lot of these dates are in the future (after the month of this table) that should not be possible

Reply:
you mean bp_date and  home_visit_date? @calellowitz .

Reply:
if you are saying about add and edd. Its possible to a to have edd in future because its expected delivery and add can be because then only definition of pregnant exisits which  is like 'if valid_in_month and add = '' or add > end of iteration month then 1 else 0'

Reply:
confirmed about edd. else definition of trimester cant exist

Reply:
reduced the edd for some of the rows. to make it closed to add to make it logically correct. But still I am pretty sure edd can be in future

Reply:
There are two dates in this file that appear in the future some rows. I cant really tell what they are, but they are the ones here shown as `2017-08-08` and `2017-03-04` (second and third added dates). The first is is the same for almost every case in the table, the latter is in the past for this case but the future (`2017-05-12`) for the case 4 down from this one. You are correct that EDD can be in the future, not sure what the other date is but i am surprised there are so many in the future.

Reply:
first one is which is `2017-08-08` is add and second one `2017-03-04` is  edd as discussed edd can be in future.
about the other date i.e `add`. I think technically 'add' can be in future for aggregation but not necessarily in the agg tables because suppose a case is opened in Aug-2018 and now is Jan-2019. She was pregnant and in Jan-2019 she delivered. So add will be say 2019-01-09. Now, when on Jan-10 2019 aggregation will run for Dec-2018 `add` would be already in future for Dec aggregation but in Dec She should be pregnant so this definition of `pregnant` handles [this](https://docs.google.com/spreadsheets/d/1RDLNIHF3iRkiQtWyrQjKH6f1bStF_S_9woXeITiMJKg/edit#gid=551271206&range=A26) condition( already in the coded in this PR)

Now to  exclude the future `add` date values from the agg table  we can set `add` only when `add` is < end of iteration month similar to date_death field here.
Does it sound reasonable?